chromatic/rainbows/actions/conversions.py

```diff
@@ -0,0 +1,79 @@
+from ...imports import *
+
+__all__ = ["to_nparray", "to_df"]
+
+
+def to_nparray(self, timeformat='h'):
+    """ Convert Rainbow object to 1D and 2D numpy arrays
+    Parameters
+    ----------
+        self : Rainbow object
+            chromatic Rainbow object to convert into array format
+        timeformat : str
+            (optional, default='hours')
+            The time format to use (seconds, minutes, hours, days etc.)
+    Returns
+    ----------
+        rflux : np.array
+            flux (MJy/sr)        [n_wavelengths x n_integrations]
+        rfluxe : np.array
+            flux error (MJy/sr)  [n_wavelengths x n_integrations]
+        rtime : np.array
+            time (BJD_TDB, houra) [n_integrations]
+        rwavel : np.array
+            wavelength (microns) [n_wavelengths]
+    """
+    secondformat = ['second', 'seconds', 'sec', 's']
+    minuteformat = ['minute', 'minutes', 'min', 'm']
+    hourformat = ['hour', 'hours', 'h']
+    dayformat = ['day', 'days', 'd']
+    yearformat = ['year', 'years', 'y']
+
+    rflux = np.array(self.fluxlike['flux'])  # flux (MJy/sr)                : [n_wavelengths x n_integrations]
+    rfluxe = np.array(self.fluxlike['uncertainty'])  # flux error (MJy/sr)  : [n_wavelengths x n_integrations]
+    rtime = np.array(self.timelike['time'])  # time (BJD_TDB, hours)        : [n_integrations]
+    rwavel = np.array(self.wavelike['wavelength'])  # wavelength (microns)  : [n_wavelengths]
+
+    # change the time array into the requested format (e.g. seconds, minutes, days etc.)
```

Comment:
For this unit conversion stuff below, I wonder would it work to do the time unit conversions using astropy? For example, would something like `self.timelike['time'].to(timeformat)` do this similarly well? Your solution works, but I could for example I could imagine someone someday looking at a dataset that spanned years; if we use astropy, it'll be super general!

Reply:
Is this the sort of thing you had in mind? I haven't used astropy.time much so I'm not sure if there's a better way?

```
from astropy.time import TimeDelta
import astropy.units as u
t = TimeDelta(24*u.hr)
print(t.to_value('s'),t.to_value('second'),t.to_value('min'),t.to_value('minute'),t.to_value('yr'),t.to_value('year'))
> 86400.0 86400.0 1440.0 1440.0 0.0027378507871321013 0.0027378507871321013
```

Reply:
I've updated this to use astropy units which is nice and clean, but my tests now fail with rounding errors:

```
t_unit='d'
rtime = self.timelike['time']  # time (BJD_TDB, hours)                : [n_integrations]

try:
        # nice bit of code copied from .imshow(), thanks Zach!
        # Change time into the units requested by the user
        t_unit = u.Unit(t_unit)
except:
        warnings.warn("Unrecognised Time Format! Returning day by default")
        t_unit = u.Unit('d')
rtime = np.array((rtime / t_unit).decompose())

rtime == r.time.to_value()/24.
>-0.10416666666666666 != -0.10416666666666667
```

Reply:
np.all(np.isclose(rtime, r.time.to_value()/24.)) is a work around but not ideal!

Reply:
Yeah, it'd be nice if it got everything right down to machine precision, but I think your `isclose` is a perfectly lovely solution. I made a few more of the `==` into `isclose` (because different ones failed on mine), and I also changed the relative tolerance (`rtol`) to be a smaller a number, so we can be more confident it's not a fluke when things match up.